tests/vmi_iothreads_test.go

```diff
@@ -299,3 +314,22 @@ func getDiskByName(domSpec *api.DomainSpec, diskName string) (*api.Disk, error)
 	}
 	return nil, fmt.Errorf("disk device '%s' not found", diskName)
 }
+
+func getFirstDiskName(vmi *v1.VirtualMachineInstance) string {
+	if len(vmi.Spec.Domain.Devices.Disks) == 0 {
+		return ""
+	}
+	return vmi.Spec.Domain.Devices.Disks[0].Name
+}
+
+// setDedicatedIOThread set disk with DedicatedIOThread = true. Assumes the disk already exist
+func setDedicatedIOThread(diskName string) libvmi.Option {
```

Comment:
shouldn't this function be in libvmi? it feels odd here.
Please consider adding a new option on libvmi `WithContainerDiskDedicatedIOThread` or something in that nature

Reply:
There are two reasons why I put it here:
1. it's only used here. I don't want to extend the common library for specific cases.
2. this Option works in a different way than the regular Options. It assumes the existence of a disk. other functions adds the missing parts, but then, if I'll implement full add disk option, we'll end with too many kind of add disk. The solution may be to add a `DiskOption` type, and a factory for disks, but I'm not sure it's really needed, and for sure it's out of the scope of this PR.

Reply:
Adding a `DiskOption` indeed may be overkill, but imo adding a specific WithContainerDisk with DedicatedIOThread is the better option. @EdDev WDYT?

Reply:
I think the nice thing about interfaces is that you can always have a local private dedicated implementation. I think here is a good example.

Reply:
It's not necessarily bad to add a local private dedicated option, but it just looked odd. Your explanation that it is only used here it good enough for me. 
However please consider keeping to the option format, like `WithContainerDiskDedicatedIOThread`.

Reply:
The different name is also on purpose, for the same reason. This option does not add disk but only modifies an existing one. I want the name to reflect the different behaviour.

Reply:
ok, renamed